docs/examples/sam_mark_duplicates.md

```diff
@@ -0,0 +1,186 @@
+---
+jupytext:
+  text_representation:
+    format_name: myst
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
+# Marking SAM read pair duplicates with oxbow
+
+```{code-cell} ipython3
+import oxbow as ox
+import polars as pl
+```
+
+## Loading the SAM File
+Let's grab a sample SAM file
+
+```{code-cell} ipython3
+# Let's use oxbow to read in the SAM file as a polars dataframe
+df = ox.from_sam("data/2133236").to_polars()
+df.head()
+```
+
+## Compute 5' start position
+
+```{code-cell} ipython3
+
+def parse_cigar(cigar_str: str) -> list[tuple[str, int]]:
+    """Parse the CIGAR string into a list of tuples (operation, length)
+
+    Example:
+        >>> parse_cigar("76M")
+        [('M', 76)]
+        >>> parse_cigar("10M1I65M")
+        [('M', 10), ('I', 1), ('M', 65)]
+    """
+    result = []
+    current_number = ""
+
+    for char in cigar_str:
+        if char.isdigit():
+            current_number += char
+        else:
+            if current_number:
+                result.append((char, int(current_number)))
+                current_number = ""
+
+    return result
+
+# If the bit 0x10 is set, the read is on the reverse strand
+STRAND_BIT = 0x10
+
+def get_unclipped_5_prime_start_position(row) -> int:
```

Comment:
This could use a correctness check as this has all been pretty new territory for me :)

Reply:
Procedurally, this looks correct to me. The only difference I see from the [reference implementation](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/SAMUtils.java#L765-L814) is that they include hard clips ("H"), whereas this seems to only account for soft clipping ("S").